src/runtime/entitlements-manager-test.js

```diff
@@ -1040,81 +1032,138 @@ describes.realWin('EntitlementsManager', {}, (env) => {
+          )
+        );
+        xhrMock
+          .expects('fetch')
+          .withExactArgs(
+            `$frontend$/swg/_/api/v1/publication/pub1/entitlements?encodedParams=${encodedParams}`,
+            {
+              method: 'GET',
+              headers: {'Accept': 'text/plain, application/json'},
+              credentials: 'include',
+            }
+          )
+          .returns(
+            Promise.resolve({
+              text: () =>
+                Promise.resolve(
+                  JSON.stringify({
+                    signedEntitlements: 'SIGNED_DATA',
+                  })
+                ),
+            })
+          );
+
+        const ents = await manager.getEntitlements({
+          metering: {
+            state: {
+              id: 'u1',
+              standardAttributes: {'att1': {timestamp: 1234567}},
+              customAttributes: {'att2': {timestamp: 1234567}},
+            },
+          },
+        });
+        expect(ents.service).to.equal('subscribe.google.com');
+        expect(ents.raw).to.equal('SIGNED_DATA');
+        expect(ents.entitlements).to.deep.equal([
+          {
+            source: 'google:metering',
             products: ['pub1:label1'],
             subscriptionToken: 'token1',
-            source: 'google:metering',
+            subscriptionTokenContents: testSubscriptionTokenContents,
+          },
+        ]);
+        expect(ents.enablesThis()).to.be.true;
+      });
+
+      it('should fetch metering entitlements with METERED_BY_GOOGLE if enabled', async () => {
+        const encodedParams = base64UrlEncodeFromBytes(
+          utf8EncodeSync(
+            '{"metering":{"clientTypes":[2,1],"owner":"pub1","resource":{"hashedCanonicalUrl":"cf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e"},"state":{"id":"u1","attributes":[{"name":"standard_att1","timestamp":1234567},{"name":"custom_att2","timestamp":1234567}]},"token":"token"}}'
+          )
+        );
+        xhrMock
+          .expects('fetch')
+          .withExactArgs(
+            `$frontend$/swg/_/api/v1/publication/pub1/entitlements?encodedParams=${encodedParams}`,
+            {
+              method: 'GET',
+              headers: {'Accept': 'text/plain, application/json'},
+              credentials: 'include',
+            }
+          )
+          .returns(
+            Promise.resolve({
+              text: () =>
+                Promise.resolve(
+                  JSON.stringify({
+                    signedEntitlements: 'SIGNED_DATA',
+                  })
+                ),
+            })
+          );
+
+        manager.enableGoogleControlledMetering();
+        const ents = await manager.getEntitlements({
+          metering: {
+            state: {
+              id: 'u1',
```

Comment:
This is allowed in this mode, but sending publisher-provided IDs or attributes in this style with METERED_BY_GOOGLE will probably be unlikely or unusual, and you could argue it's not supported at this time. Perhaps remove?

Reply:
We can't exercise this code-path unless a state is provided

Reply:
I think you're right that we shouldn't explicitly support this case today. Will just replace if a state was provided